proc/groupby.go

```diff
@@ -135,59 +147,44 @@ type GroupByAggregator struct {
 	keys        []GroupByKey
 	reducerDefs []compile.CompiledReducer
 	builder     *ColumnBuilder
-	// For a regular group-by, tables has one entry with key 0.  For a
-	// time-binned group-by, tables has one entry per bin and is keyed by
-	// bin timestamp (so that a bin with span [ts, ts+timeBinDuration) has
-	// key ts).
-	tables          map[nano.Ts]map[string]*GroupByRow
-	TimeBinDuration int64 // Zero means regular group-by (no time binning).
-	reverse         bool
-	logger          *zap.Logger
-	limit           int
+	table       map[string]*GroupByRow
+	reverse     bool
+	logger      *zap.Logger
+	limit       int
 }
 
 type GroupByRow struct {
 	keycols  []zng.Column
 	keyvals  zcode.Bytes
-	ts       nano.Ts
 	reducers compile.Row
 }
 
 func NewGroupByAggregator(c *Context, params GroupByParams) *GroupByAggregator {
-	//XXX we should change this AST format... left over from Looky
-	// convert second to nano second
-	dur := int64(params.duration.Seconds) * 1000000000
-	if dur < 0 {
-		panic("dur cannot be negative")
-	}
 	limit := params.limit
 	if limit == 0 {
 		limit = defaultGroupByLimit
 	}
 	return &GroupByAggregator{
-		keys:            params.keys,
-		zctx:            c.TypeContext,
-		kctx:            resolver.NewContext(),
-		reducerDefs:     params.reducers,
-		builder:         params.builder,
-		keyRows:         make(map[int]keyRow),
-		tables:          make(map[nano.Ts]map[string]*GroupByRow),
-		TimeBinDuration: dur,
-		reverse:         c.Reverse,
-		logger:          c.Logger,
-		limit:           limit,
+		keys:        params.keys,
+		zctx:        c.TypeContext,
+		kctx:        resolver.NewContext(),
+		reducerDefs: params.reducers,
+		builder:     params.builder,
+		keyRows:     make(map[int]keyRow),
+		table:       make(map[string]*GroupByRow),
+		reverse:     c.Reverse,
```

Comment:
Do we still need a reverse flag with these changes?

Reply:
The reverse flag wasn't properly handled (#847) and per the comment my proposal (to be debated) is to let groupby output time bins in "unsorted  but deterministic" order like for other keys...